File: Ser2023/Store/Store.Web/Modules/ServerTypes/Store/CategoriesExcelImportForm.ts
import { ImageUploadEditor, PrefixedContext } from "@serenity-is/corelib";
import { initFormType } from "@serenity-is/corelib/q";

export interface CategoriesExcelImportForm {
    FileName: ImageUploadEditor;
}

export class CategoriesExcelImportForm extends PrefixedContext {
    static formKey = 'Store.CategoriesExcelImport';
    private static init: boolean;

    constructor(prefix: string) {
        super(prefix);

        if (!CategoriesExcelImportForm.init)  {
            CategoriesExcelImportForm.init = true;

            var w0 = ImageUploadEditor;

            initFormType(CategoriesExcelImportForm, [
                'FileName', w0
            ]);
        }
    }
}

File: Ser2023/Categories/CategoriesExcelImportForm.cs

namespace Store.Store.Forms
{
    using Serenity.ComponentModel;
    using Serenity.Web;
    using System;

    [FormScript("Store.CategoriesExcelImport")]
    public class CategoriesExcelImportForm
    {
        [FileUploadEditor, Required]
        public String FileName { get; set; }
    }
}